stdlib/test/shutil/test_shutil.mojo

```diff
@@ -0,0 +1,54 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+# RUN: %mojo %s
+
+from shutil import rmtree
+from os import mkdir
+from os.path import exists
+from pathlib import Path
+
+from testing import assert_false, assert_raises
+
+
+fn _create_some_files_and_dirs(path: Path) raises:
+    mkdir(path / "nested")
+    mkdir(path / "nested" / "more_nested")
+    mkdir(path / "nested" / "more_nested" / "even_more_nested")
+
+    _ = open(path / "nested.txt", "w")
+    _ = open(path / "nested" / "more_nested.txt", "w")
+    _ = open(path / "nested" / "more_nested" / "even_more_nested.txt", "w")
+
+
+fn test_rmtree() raises:
+    var cwd_path = Path()
+    var my_dir_path = cwd_path / "my_dir"
```

Comment:
I'm not so sure about this. It would be nice if we had some `tempfile`-like utilities to create these ephemeral paths. At the same time, `tempfile` should probably depend on `shutil`, not the other way around.
Idea 1: Maybe we could bring up some basic `tempfile` utilities so that we can at least have a platform-independent temporary path.
Idea 2: we could just use python interop in these tests while we get there.

Reply:
I guess python interop is the best option for now, one idea to prevent testing `shutil` with a potential `tempfile` module (which logically would depend on `shutil`) would be to use something like `subprocess.run` and call something like `rm -rf my_dir`, but it does not seem like a short-term solution